src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -2877,24 +2878,28 @@ export class ResultsViewPageStore {
 
     });
 
-    public readonly sampleGroups = remoteData({
+    public readonly sampleGroups = remoteData<SampleGroup[]>({
         // not sure how we'll get these from the server, if we'll have to process or fetch more
         // for now let's just do this for testing purposes, and it won't be hard to switch around
         await:()=>[this.samples, this.alteredSamples, this.unalteredSamples],
         invoke:()=>{
             return Promise.resolve([
                 {
                     name: "Altered Samples",
-                    sampleIdentifiers: this.alteredSamples.result!.map(s=>({ sampleId: s.sampleId, studyId: s.studyId }))
+                    sampleIdentifiers: this.alteredSamples.result!.map(s=>({ sampleId: s.sampleId, studyId: s.studyId })),
+                    color: "red",
+                    legendText: "Cases with Alteration(s) in Query Gene(s)"
                 },
                 {
                     name: "Unaltered Samples",
-                    sampleIdentifiers: this.unalteredSamples.result!.map(s=>({ sampleId: s.sampleId, studyId: s.studyId }))
+                    sampleIdentifiers: this.unalteredSamples.result!.map(s=>({ sampleId: s.sampleId, studyId: s.studyId })),
+                    color: "blue",
+                    legendText: "Cases without Alteration(s) in Query Gene(s)"
                 },
-                {
+                /* {
```

Comment:
wait why have you commented this out?

Reply:
this group contains all the samples from first two groups. With the existing logic in SurvivaChart only this group would be drawn. i.e, it doesn't handle overlapping samples/patients. Not sure how to deal with it.

@jjgao How should we deal with groups with overlapping samples in survival plots?

Reply:
that seems like a code obstacle not a conceptual obstacle, right?

Reply:
even conceptual obstacle. whether to show same sample in multiple lines.

Reply:
may be for now i'll show it